detectable/src/main/java/com/synopsys/integration/detectable/detectables/carthage/CarthageExtractor.java

```diff
@@ -1,30 +1,32 @@
 package com.synopsys.integration.detectable.detectables.carthage;
 
 import java.io.File;
+import java.io.IOException;
 import java.nio.file.Files;
 import java.util.List;
 
 import com.synopsys.integration.bdio.graph.DependencyGraph;
 import com.synopsys.integration.detectable.detectable.codelocation.CodeLocation;
+import com.synopsys.integration.detectable.detectables.carthage.model.CarthageDeclaration;
+import com.synopsys.integration.detectable.detectables.carthage.parse.CartfileResolvedParser;
+import com.synopsys.integration.detectable.detectables.carthage.transform.CarthageDeclarationTransformer;
 import com.synopsys.integration.detectable.extraction.Extraction;
 
 public class CarthageExtractor {
-    private CartfileResolvedDependencyDeclarationParser dependencyDeclarationParser;
+    private final CartfileResolvedParser cartfileResolvedParser;
+    private final CarthageDeclarationTransformer declarationTransformer;
 
-    public CarthageExtractor(CartfileResolvedDependencyDeclarationParser dependencyDeclarationParser) {
-        this.dependencyDeclarationParser = dependencyDeclarationParser;
+    public CarthageExtractor(CartfileResolvedParser cartfileResolvedParser, CarthageDeclarationTransformer declarationTransformer) {
+        this.cartfileResolvedParser = cartfileResolvedParser;
+        this.declarationTransformer = declarationTransformer;
     }
 
-    public Extraction extract(File cartfileResolved) {
-        try {
-            List<String> dependencyDeclarations = Files.readAllLines(cartfileResolved.toPath());
-
-            DependencyGraph dependencyGraph = dependencyDeclarationParser.parseDependencies(dependencyDeclarations);
-            CodeLocation codeLocation = new CodeLocation(dependencyGraph);
-            // No project info - hoping git can help with that.
-            return new Extraction.Builder().success(codeLocation).build();
-        } catch (Exception e) {
-            return new Extraction.Builder().failure(String.format("There was a problem extracting dependencies from %s", cartfileResolved.getAbsolutePath())).build();
-        }
+    public Extraction extract(File cartfileResolved) throws IOException {
+        List<String> dependencyDeclarations = Files.readAllLines(cartfileResolved.toPath());
+        List<CarthageDeclaration> carthageDeclarations = cartfileResolvedParser.parseDependencies(dependencyDeclarations);
+        DependencyGraph dependencyGraph = declarationTransformer.transform(carthageDeclarations);
+        CodeLocation codeLocation = new CodeLocation(dependencyGraph);
+        // No project info - hoping git can help with that.
+        return new Extraction.Builder().success(codeLocation).build();
```

Comment:
Not necessarily related to the PR :) but how would we feel about having static methods for the really common extraction types? Extraction.success(codeLocation) so if you don't do anything advanced you could just use that instead?

Reply:
Makes sense to me, I'll do it here